### PR TITLE
astro-language-server: 2.10.0 -> 2.14.2

### LINKED ISSUE #343202: Update request: astro-language-server 2.10.0 -> 2.14.2
- Package name: astro-language-server
- Latest released version: 2.14.2

- Current version on the unstable channel: 2.10.0
- Current version on the stable/release channel: None

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**

-----

Note for maintainers: Please tag this issue in your PR.

---
@pyrox0 

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by DataHearth**
## Description of changes

- Updated to latest version
- Added a pnpm build dependency due to `@astrojs/yaml2ts` behind linked with the pnpm workspace in lockfile

Closes #343202

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by DataHearth** [comment]
Result of `nixpkgs-review pr 343225` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>astro-language-server</li>
  </ul>
</details>

File: pkgs/by-name/as/astro-language-server/package.nix [hunk 1]
```diff
@@ -35,15 +35,20 @@ stdenv.mkDerivation (finalAttrs: {
 
   buildInputs = [ nodejs_22 ];
 
-  pnpmWorkspace = "@astrojs/language-server";
+  # Must specify to download dependencies for its dependencies (required for worspace with yaml2ts)
```
Comment (MisileLab) [inline pkgs/by-name/as/astro-language-server/package.nix:38]: I think this comment has a typo error (required for worspace -> required for workspace)

**Review by MisileLab — CHANGES_REQUESTED** [review]

File: pkgs/by-name/as/astro-language-server/package.nix (hunk 1 above)
Comment (DataHearth) [reply pkgs/by-name/as/astro-language-server/package.nix:38]: absolutely... I'm going to fix ^^. Ty

**Review by pyrox0 — APPROVED** [review]

File: pkgs/by-name/as/astro-language-server/package.nix [hunk 2]
```diff
@@ -61,6 +67,7 @@ stdenv.mkDerivation (finalAttrs: {
   meta = {
     description = "The Astro language server";
     homepage = "https://github.com/withastro/language-tools";
+    changelog = "https://github.com/withastro/language-tools/blob/main/packages/language-server/CHANGELOG.md";
```
Comment (gepbird) [inline pkgs/by-name/as/astro-language-server/package.nix:70]:
```suggestion
    changelog = "https://github.com/withastro/language-tools/blob/@astrojs/language-server@${finalAttrs.version}/packages/language-server/CHANGELOG.md";
```

**Review by gepbird — CHANGES_REQUESTED** [review]
Tested on x86_64-linux, works well.

**Review by gepbird — APPROVED** [review]

**Review by pyrox0 — APPROVED** [review]
My review hasn't changed, good to merge.

**Comment by DataHearth** [comment]
Any news from a maintainer with write permission ??

**Comment by gepbird** [comment]
cc @doronbehar

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 343225` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>astro-language-server</li>
  </ul>
</details>

**Pull request merged**